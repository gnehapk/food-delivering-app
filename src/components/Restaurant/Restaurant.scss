.restaurant{
  margin: 3rem 3rem 0 3rem;
  width: 360px;
}