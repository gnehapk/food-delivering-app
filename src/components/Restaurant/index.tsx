import Restuarant from './Restaurant';

export default Restuarant;