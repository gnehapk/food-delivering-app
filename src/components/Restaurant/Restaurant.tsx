import React from 'react';
import RestaurantImg from '../RestaurantImg';
import RestaurantDetails from '../RestaurantDetails';
import { RestaurantType } from '../RestuarantListBuilder/RestaurantListBuilder';

import './Restaurant.scss';

const Restuarant: React.FC<RestaurantPropsType> = (props) => {
  return (
    <div className="restaurant">
      <RestaurantImg restaurantImg={props.details.image}/>
      <RestaurantDetails details={props.details}/>
    </div>
  );
};

type RestaurantPropsType = {
  details: RestaurantType;
};

export default Restuarant;