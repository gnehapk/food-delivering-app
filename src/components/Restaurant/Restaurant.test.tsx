import React from 'react';
import TestRenderer from 'react-test-renderer';
import Restaurant from './Restaurant';

describe('Restuarant', () => {
  it('should render correctly', () => {
    const details = {
      id: '1',
      name: 'Kitchen',
      image: 'http://kitchen.png',
      url: 'http://kitchen.png',
      price: 100,
      tags: ['tag1', 'tag2'],
    };
    const tree = TestRenderer.create(<Restaurant details={details} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

