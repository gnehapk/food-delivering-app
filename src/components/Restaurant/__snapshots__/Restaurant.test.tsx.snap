// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restuarant should render correctly 1`] = `
<div
  className="restaurant"
>
  <img
    alt="RestaurantImg"
    src="http://kitchen.png"
  />
  <div>
    <div
      className="restuarant-name"
    >
      Kitchen
    </div>
    <div
      className="tags"
    >
      <div
        className="tag-list"
      >
        <div
          className="tag"
        >
           
          tag1
        </div>
        <div
          className="tag"
        >
           
          tag2
        </div>
      </div>
      <div>
        £ 
        100
      </div>
    </div>
  </div>
</div>
`;
